.github/workflows/ci-pr.yml

```diff
@@ -0,0 +1,119 @@
+name: CI-PR
+on:
+  pull_request:
+    branches: [master]
+concurrency:
+  group: ci-${{ github.head_ref }}
+  cancel-in-progress: true
+jobs:
+  ci-pr:
+    name: CI-PR
+    needs: [test, msrv, lint]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Done
+        run: exit 0
+  test:
+    name: Tests
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ["ubuntu-latest", "windows-latest"]
+        rust: ["stable"]
```

Comment:
IMHO this should always be `1.54.0` because that is what we should be developing in.

Reply:
I had considered this, just not testing stable at all.  However, I remembered when a dependency update went past our MSRV.  That doesn't automatically force us to update our MSRV but we aren't prepared with a strategy for when that happens.  Keeping the majority of CI on `stable` means most things will work until we do come up with a workaround.

Reply:
Moving this comment to keep the conversation in one place

> > That doesn't automatically force us to update our MSRV but we aren't prepared with a strategy for when that happens
> 
> We need to know when that happens. The main reasons I want msrv here are:
> 
>     1. stable is being tested on master.
> 
>     2. MSRV is most important thing people would miss when contributing because as you said, they would probably develop on stable and thus getting it work on stable.

Reply:
> > That doesn't automatically force us to update our MSRV but we aren't prepared with a strategy for when that happens

> We need to know when that happens

When designing pipelines, one of the things I consider is "is this sending the right message to the right person"?  MSRV failures caused by a release of a dependency is unrelated to existing PRs and we should not block progress on a PR because of it

> MSRV is most important thing people would miss when contributing

In my experience, 99% of MSRV problems are compilation problems and not runtime problems.

Reply:
Forgot to add, we'll still get the runtime feedback when bors does the full build before merging into master.  So we are still covered for corner cases.

Reply:
>  MSRV failures caused by a release of a dependency is unrelated to existing PRs and we should not block progress on a PR because of it

Agreed, however they can also happen due to someone submitting a PR that utilizes features not present in the MSRV. It's happened a few times before, but those are all *compile errors* not runtime errors. Which brings us to the next comment :smile: 

> In my experience, 99% of MSRV problems are compilation problems and not runtime problems

100% agree. I'd be hard pressed to think of a circumstance where the change results in MSRV still compiling but failing a test. I think due to Rust's backwards compatibility guarantees this should be *almost* a non-issue.